batch/job_launcher.py

```diff
@@ -0,0 +1,103 @@
+              help="total number of jobs to run in this batch")
+@click.option("-s", "--sims-per-job", "sims_per_job", type=click.IntRange(min=1), required=True,
+              help="how many sims each job should run")
+@click.option("-t", "--dvc-target", "dvc_target", type=click.Path(exists=True), required=True,
+              help="name of the .dvc file that is the last step in the pipeline")
+@click.option("-i", "--s3-input-bucket", "s3_input_bucket", type=str, default="idd-input-data-sets")
+@click.option("-o", "--s3-output-bucket", "s3_output_bucket", type=str, default="idd-pipeline-results")
+@click.option("-d", "--job-definition", "batch_job_definition", type=str, default="Batch-CovidPipeline-Job")
+@click.option("-q", "--job-queue", "batch_job_queue", type=str, default="Batch-CovidPipeline")
+def launch_batch(job_prefix, config_file, num_jobs, sims_per_job, dvc_target, s3_input_bucket, s3_output_bucket, batch_job_definition, batch_job_queue):
+
+    # A unique name for this job run, based on the job prefix and current time
+    job_name = "%s-%d" % (job_prefix, int(time.time()))
+    print("Preparing to run job: %s" % job_name)
+
+    # Update and save the config file with the number of sims to run
+    print("Updating config file %s to run %d simulations..." % (config_file, sims_per_job))
+    config = open(config_file).read()
+    config = re.sub("nsimulations: \d+", "nsimulations: %d" % sims_per_job, config)
+    with open(config_file, "w") as f:
+        f.write(config)
+
+    # Prepare to tar up the current directory, excluding any dvc outputs, so it
+    # can be shipped to S3
+    dvc_outputs = get_dvc_outputs()
+    tarfile_name = "%s.tar.gz" % job_name
+    tar = tarfile.open(tarfile_name, "w:gz")
+    for p in os.listdir('.'):
+        if not (p.startswith(".") or p.endswith("tar.gz") or p in dvc_outputs or p == "batch"):
+            tar.add(p)
+    tar.close()
+ 
+    # Upload the tar'd contents of this directory and the runner script to S3 
+    runner_script_name = "%s-runner.sh" % job_name
+    s3_client = boto3.client('s3')
+    s3_client.upload_file("batch/runner.sh", s3_input_bucket, runner_script_name)
+    s3_client.upload_file(tarfile_name, s3_input_bucket, tarfile_name)
+    os.remove(tarfile_name)
+
+    # Prepare and launch the num_jobs via AWS Batch.
+    model_data_path = "s3://%s/%s" % (s3_input_bucket, tarfile_name)
+    results_path = "s3://%s/%s" % (s3_output_bucket, job_name)
+    env_vars = [
+            {"name": "CONFIG_PATH", "value": config_file},
+            {"name": "S3_MODEL_DATA_PATH", "value": model_data_path},
+            {"name": "DVC_TARGET", "value": dvc_target},
+            {"name": "DVC_OUTPUTS", "value": " ".join(dvc_outputs)},
+            {"name": "S3_RESULTS_PATH", "value": results_path}
+    ]
+    s3_cp_run_script = "aws s3 cp s3://%s/%s $PWD/run-covid-pipeline" % (s3_input_bucket, runner_script_name)
```

Comment:
(aside, don't know if you know about fstrings in python 3.6+, they make subbing variables much easier/neater)

Reply:
I do not really; will take a look!